### PR TITLE
Fall back to simple text output when stdout is not a TTY

### LINKED ISSUE #113: Commands like `image pull` don't output any status if stdout (or stderr) is being piped to a file
If you run `container image pull` with stdout (or stderr) piped to a file, there's no progress output from the command. Ideally commands could fall back to a simple stream of status output if running without a terminal. This would make it easier to use the new CLI from other applications.

### PR DESCRIPTION
**Pull request by knQzx**
## summary
- detect when stdout is not a tty and fall back to simple line-by-line status output
- fixes silent commands when piping to a file

fixes #113

## test plan

**before (main):**
```
$ container image pull ghcr.io/linuxcontainers/alpine:3.20 2>stderr.txt
$ cat stderr.txt
(empty - progress lost when stderr is not a tty)
```

**after:**
```
$ container image pull ghcr.io/linuxcontainers/alpine:3.20 2>stderr.txt
$ cat stderr.txt
[1/2] Fetching image [0s]
[1/2] Fetching image (7 of 15 blobs) [2s]
[1/2] Fetching image 42% (23 of 50 blobs, 10.0/23.7 MB, 6.9 MB/s) [5s]
[2/2] Unpacking image for platform linux/arm64 [11s]
...
```

also added `TestCLIProgressAuto` integration test covering auto/plain/none modes

### CODE REVIEW

**Comment by jglogan** [comment]
@knQzx Thank you for the PR! Could you please add the commands you tested with, and before/after output, to the PR description?

Would it be possible to add an integration test for this?

**Comment by knQzx** [comment]
added before/after output to the description and an integration test for this

**Comment by jglogan** [comment]
@knQzx It was a long time ago, but I think we made an implementation decision during initial progress bar development that if stderr wasn't a tty, we would suppress progress output, so that you'd only see warnings and errors.

I can think of an idea or two, but why is this change to include progress with the warnings and errors useful for you?

**Comment by knQzx** [comment]
@jglogan mainly for ci logs - when a pull runs in a pipeline and stderr goes to a file, it's hard to tell if things are progressing or stuck. what were the ideas you had in mind?

**Comment by jglogan** [comment]
@knQzx I think it's similar.

The use case I was thinking about originally was, if run `container something ... 2> errors.txt` I really want to study the errors/warnings, so progress just gets in the way for me.

OTOH, if that command is long running and I want to do `tail -f errors.txt`, being able to see that a command is making progress / doesn't appear to be stuck is probably exactly why I'm running that `tail`.

@dkovba Do you remember anything else from when you were developing the progress bar and we were talking about it? Looking at where we are with `container` now, do you have a preference?

**Comment by jglogan** [comment]
@knQzx Could you review https://github.com/apple/containerization/blob/main/CONTRIBUTING.md#submitting-issues-and-pull-requests and set up signing for your commits? They need to show up in the PR as `Verified`, like:

image

Then the CI build can make progress.

File: Tests/CLITests/Subcommands/Images/TestCLIProgressAuto.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import Testing
+
+class TestCLIProgressAuto: CLITest {
+    @Test func testAutoProgressFallsBackToPlainWhenPiped() throws {
+        do {
```
Comment (dkovba) [inline Tests/CLITests/Subcommands/Images/TestCLIProgressAuto.swift]: `do-catch` blocks in tests seem redundant.

File: Sources/ContainerCommands/Flags+ProgressConfig.swift [hunk 2]
```diff
@@ -15,14 +15,31 @@
...
 import TerminalProgress
 
 extension Flags.Progress {
+    /// Resolves `.auto` into `.ansi` or `.plain` based on the terminal.
+    ///
+    /// When the progress type is `.auto`, the terminal file handle is checked
+    /// with `isatty`. If the output is a TTY, `.ansi` is used; otherwise
+    /// `.plain` is selected so that piped or redirected output still shows
+    /// simple line-by-line status text.
+    private func resolvedProgress(terminal: FileHandle = .standardError) -> ProgressType {
```
Comment (dkovba) [inline Sources/ContainerCommands/Flags+ProgressConfig.swift]: The `terminal` parameter is currently unused. Do we need it? Also, the default value `.standardError` contradicts the PR title mentioning `stdout`.

File: Tests/CLITests/Subcommands/Images/TestCLIProgressAuto.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            Issue.record("failed to test plain progress: \(error)")
+            return
+        }
+    }
+
+    @Test func testNoneProgressSuppressesOutput() throws {
+        do {
+            let (_, _, error, status) = try run(arguments: [
+                "image", "pull",
+                "--progress", "none",
```
Comment (dkovba) [inline Tests/CLITests/Subcommands/Images/TestCLIProgressAuto.swift]: Consider adding the same test for `--progress ansi`.

**Comment by dkovba** [comment]
Thank you for the contribution! 👍

**Comment by knQzx** [comment]
@jglogan commits are signed now, also addressed @dkovba's review feedback - removed the unused terminal param, cleaned up the test do-catch blocks, and added an ansi progress test

**Comment by jglogan** [comment]
@knQzx Looks like yours is building now, thanks!

You'll need to resolve some conflicts since the `--progress color` PR is merged now.

**Comment by knQzx** [comment]
@jglogan rebased, conflicts resolved

**Review by jglogan — APPROVED** [review]

**Comment by jglogan** [comment]
@knQzx Thank you for the new feature, merged!
